subsys/bluetooth/scan.c

```diff
@@ -791,6 +814,104 @@ static int scan_appearance_filter_add(u16_t appearance)
 	return 0;
 }
 
+static bool adv_manufacturer_data_cmp(const u8_t *data,
+		u8_t data_len,
+		const u8_t *target_data,
+		u8_t target_data_len)
+{
+	if (target_data_len > data_len) {
+		return false;
+	}
+
+	for (size_t i = 0; i < target_data_len; i++) {
+		if (target_data[i] != data[i]) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static bool adv_manufacturer_data_compare(const struct bt_data *data)
+{
+	struct bt_scan_manufacturer_data_filter const *md_filter =
+		&bt_scan.scan_filters.manufacturer_data;
+	u8_t counter = bt_scan.scan_filters.manufacturer_data.cnt;
+
+	/* Compare the name found with the name filter. */
+	for (size_t i = 0; i < counter; i++) {
+		if (adv_manufacturer_data_cmp(data->data,
+				data->data_len,
+				md_filter->manufacturer_data[i].data,
+				md_filter->manufacturer_data[i].data_len)) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static bool is_manufacturer_data_filter_enabled(void)
+{
+	return bt_scan.scan_filters.manufacturer_data.enabled;
+}
+
+static void manufacturer_data_check(struct bt_scan_control *control,
+		const struct bt_data *data)
+{
+	if (is_manufacturer_data_filter_enabled()) {
+		if (adv_manufacturer_data_compare(data)) {
+			control->filter_match_cnt++;
+
+			/* Information about the filters matched. */
+			control->filter_status.manufacturer_data = true;
+			control->filter_match = true;
+		}
+	}
+}
+
+static int scan_manufacturer_data_filter_add(
+		const struct bt_scan_manufacturer_data *manufacturer_data)
```

Comment:
Please move to the previous line.

Reply:
But if I move it to the previous line, the line will be >80 chars?

Reply:
Yes, but when the line is > 80 chars it causes checkpatch warning not an error, so in case when exceeding the line length increase readability of the code I would prefer to break this rule.
So you should not exceed 80 chars per line only when it does not "spoil readability"